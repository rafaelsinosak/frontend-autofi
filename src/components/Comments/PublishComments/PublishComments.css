.inputField {
  width: 100%;
  height: 100%;
}

.buttonPublish {
  height: 100%;
  width: 100%;
}

form {
  margin-bottom: 15px;
}
