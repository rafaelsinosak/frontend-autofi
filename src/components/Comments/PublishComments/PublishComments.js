import { Grid, Button, TextField, Container } from "@material-ui/core";
import SendIcon from '@material-ui/icons/Send';

import "./PublishComments.css";

const PublishComments = (props) => {
  const { comment, onChange, onSubmit } = props;
  return (
    <Container>
      <form noValidate onSubmit={onSubmit}>
        <Grid
          container
          alignItems="center"
          direction="row"
          justifyContent="center"
        >
          <Grid item xs={11}>
            <TextField
              id="outlined-basic"
              label="Comment"
              variant="outlined"
              className="inputField"
              value={comment}
              onChange={onChange}
            />
          </Grid>
          <Grid item xs={1}>
            <Button
              size="large"
              variant="contained"
              color="primary"
              type="submit"
              disabled={!comment}
            >
              <SendIcon/>
            </Button>
          </Grid>
        </Grid>
      </form>
    </Container>
  );
};

PublishComments.defaultProps = {
  onChange: () => {},
  onSubmit: () => {},
};

export default PublishComments;
