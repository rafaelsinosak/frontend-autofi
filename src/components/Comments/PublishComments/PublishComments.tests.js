import React from 'react';
import renderer from 'react-test-renderer';
import PublishComments from './PublishComments';

test('PublishComments - Render Component', () => {
  const component = renderer.create(
    <PublishComments></PublishComments>,
  );
  let publishComments = component.toJSON();
  expect(publishComments).toMatchSnapshot();
});