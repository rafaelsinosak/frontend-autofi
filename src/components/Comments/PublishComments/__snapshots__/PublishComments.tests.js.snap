// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PublishComments - Render Component 1`] = `
<div
  className="MuiContainer-root MuiContainer-maxWidthLg"
>
  <form
    noValidate={true}
    onSubmit={[Function]}
  >
    <div
      className="MuiGrid-root MuiGrid-container MuiGrid-align-items-xs-center MuiGrid-justify-content-xs-center"
    >
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-11"
      >
        <div
          className="MuiFormControl-root MuiTextField-root inputField"
        >
          <label
            className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
            data-shrink={false}
            htmlFor="outlined-basic"
            id="outlined-basic-label"
          >
            Comment
          </label>
          <div
            className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
            onClick={[Function]}
          >
            <input
              aria-invalid={false}
              autoFocus={false}
              className="MuiInputBase-input MuiOutlinedInput-input"
              disabled={false}
              id="outlined-basic"
              onAnimationStart={[Function]}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              required={false}
              type="text"
            />
            <fieldset
              aria-hidden={true}
              className="PrivateNotchedOutline-root-1 MuiOutlinedInput-notchedOutline"
            >
              <legend
                className="PrivateNotchedOutline-legendLabelled-3"
              >
                <span>
                  Comment
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
      </div>
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-1"
      >
        <button
          className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-containedSizeLarge MuiButton-sizeLarge Mui-disabled Mui-disabled"
          disabled={true}
          onBlur={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={-1}
          type="submit"
        >
          <span
            className="MuiButton-label"
          >
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M2.01 21L23 12 2.01 3 2 10l15 2-15 2z"
              />
            </svg>
          </span>
        </button>
      </div>
    </div>
  </form>
</div>
`;
